### ISSUE TITLE
Incorrect detection of async functions

### ISSUE THREAD

**Issue by fxOne** [SonarSource/sonar-javascript#727]
**I want to report a bug.**

**SonarJS version:** 3.1.1<br>
**SonarQube version:** Version 6.5 (build 27846)<br>

**Rule key:** javascript:Semicolon

**Reproducer** 
```
async function outerFunction() {
  async function innerAsyncFunction() {

  }
  function innerSyncFunction() {

  }
  innerSyncFunction();
  await innerAsyncFunction();
}
```
I'm getting the error `Add a semicolon at the end of this statement.` for the `innerAsyncFunction` function.

**Expected behavior**
I expected no error as it makes no sense to add an semicolon at the end of the async function definition. I don't get the ticket for the `outerFunction` or `innerSyncFunction` functions.

**Comment by vilchik-elena** [comment]
Nice catch, @fxOne thanks for reporting!

Looks like we are mistakenly parsing `innerAsyncFunction` as function expression, while it should be function declaration.

**Comment by inverno** [comment]
@fxOne fix is now part of : https://github.com/SonarSource/sonar-javascript/releases/tag/3.2-RC1
